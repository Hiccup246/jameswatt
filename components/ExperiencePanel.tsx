export default function ExperiencePanel({
  job,
  activated,
}: {
  job: Job;
  activated: boolean;
}) {
  const activatedClasses = "pt-2.5 visible relative opacity-100 activated";
  const deactivatedClasses = "opacity-0 invisible";
  const tabPanelClasses = `absolute m-0 p-0 w-full transition-opacity duration-700 ease-in-out top-0 ${
    activated ? activatedClasses : deactivatedClasses
  }`;

  return (
    <div className={tabPanelClasses}>
      <h2 className="m-0">
        <span className="text-xl font-bold text-black">{job.role}</span>
        <span className="text-xl font-bold text-black"> @ {job.company}</span>
      </h2>
      <p className="mt-1 mb-4 text-black">{job.dateRange}</p>
      <div className="pb-4">{job.description}</div>
      <ul className="m-0 list-none p-0">
        {job.jobHighlights.map((highlight: string, index: number) => {
          return (
            <li
              className={`
                    relative mt-4 pl-8 pb-2.5 before:absolute
                    before:left-0 before:text-black before:content-["â†’"]
                  `}
              key={index}
              dangerouslySetInnerHTML={{ __html: highlight }}
            ></li>
          );
        })}
      </ul>
    </div>
  );
}
