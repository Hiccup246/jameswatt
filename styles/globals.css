@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --slider-items: 7;
    --slider-item-sets: 2;
    --all-slider-items: calc(var(--slider-items) * var(--slider-item-sets));
    --slider-item-width: 200px;
  }

  html,
  body {
    padding: 0;
    margin: 0;
    font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
      "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif,
      "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol",
      "Noto Color Emoji";
    @apply bg-brown;
  }

  @property --gradient-angle {
    syntax: "<angle>";
    inherits: false;
    initial-value: 1turn;
  }
}

@layer utilities {
  .animation-pause {
    animation-play-state: paused;
  }

  .animation-play {
    animation-play-state: running;
  }
}
