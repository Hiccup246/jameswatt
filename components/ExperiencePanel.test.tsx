import { expect, test } from "@jest/globals";
import ExperiencePanel from "./ExperiencePanel";
import renderer from "react-test-renderer";

const dummyJob: Job = {
  role: "Head Mouser",
  company: "Dummy 1",
  dateRange: "2020/09 - 2022/10",
  description: "My first dummy job",
  jobHighlights: ["highlight 1", "highlight 2"],
};

test("renders the ExperiencePanel correctly", () => {
  const tree = renderer
    .create(<ExperiencePanel job={dummyJob} activated={true} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
