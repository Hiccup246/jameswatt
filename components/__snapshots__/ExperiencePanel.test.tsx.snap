// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ExperiencePanel correctly 1`] = `
<div
  className="absolute m-0 p-0 w-full transition-opacity duration-700 ease-in-out top-0 pt-2.5 visible relative opacity-100 activated"
>
  <h2
    className="m-0"
  >
    <span
      className="text-xl font-bold text-black"
    >
      Head Mouser
    </span>
    <span
      className="text-xl font-bold text-black"
    >
       @ 
      Dummy 1
    </span>
  </h2>
  <p
    className="mt-1 mb-4 text-black"
  >
    2020/09 - 2022/10
  </p>
  <div
    className="pb-4"
  >
    My first dummy job
  </div>
  <ul
    className="m-0 list-none p-0"
  >
    <li
      className="
                    relative mt-4 pl-8 pb-2.5 before:absolute
                    before:left-0 before:text-black before:content-["→"]
                  "
      dangerouslySetInnerHTML={
        {
          "__html": "highlight 1",
        }
      }
    />
    <li
      className="
                    relative mt-4 pl-8 pb-2.5 before:absolute
                    before:left-0 before:text-black before:content-["→"]
                  "
      dangerouslySetInnerHTML={
        {
          "__html": "highlight 2",
        }
      }
    />
  </ul>
</div>
`;
